train.py

```diff
@@ -98,12 +99,13 @@ def train(args):
                 for i, (c, h) in enumerate(model.initial_state):
                     feed[c] = state[i].c
                     feed[h] = state[i].h
-                train_loss, state, _ = sess.run([model.cost, model.final_state, model.train_op], feed)
+                train_loss, state, _, summary = sess.run([model.cost, model.final_state, model.train_op, model.summary_op], feed)
                 end = time.time()
                 print("{}/{} (epoch {}), train_loss = {:.3f}, time/batch = {:.3f}" \
                     .format(e * data_loader.num_batches + b,
                             args.num_epochs * data_loader.num_batches,
                             e, train_loss, end - start))
+                writer.add_summary(summary, e)
```

Comment:
Whoops this should actually be `e * data_loader.num_batches + b`, and since that gets used several places now, should pull it out into a variable